backend/services/interfaces/testSessionService.ts

```diff
@@ -41,14 +41,28 @@ export interface TestSessionResponseDTO {
    * the result of the test session
    * there should be one entry here per student
    * */
-  results?: ResultResponseDTO[];
+  results?: GradedResultResponseDTO[];
   /** the code that students can use to access the test when it is live */
   accessCode: string;
   /** the time when the test session is started by teacher */
   startTime: Date;
 }
 
-export interface ResultRequestDTO {
+/**
+ * This interface contains the request object that is fed into the test session
+ * service that represents a new test result that has not yet been graded
+ */
+export interface UngradedResultDTO {
```

Comment:
i'm wondering why we need to separate result objects? can we not just have one with the `breakdown` and `score` set to null or empty?

Reply:
we could potentially even add a `status` field and have it default to `ungraded` and then set it to `graded` once it is.

Reply:
i like this suggestion! thanks